lib/ProductOpener/APIProductWrite.pm

```diff
@@ -50,6 +51,38 @@ use ProductOpener::Lang qw/:all/;
 use ProductOpener::Products qw/:all/;
 use ProductOpener::API qw/:all/;
 use ProductOpener::Packaging qw/:all/;
+use ProductOpener::Text qw/:all/;
+
+use Encode;
+
+=head2 skip_protected_field($product_ref, $field, $moderator = 0)
+
+Return 1 if we should ignore a field value sent by an user because we already have a value sent by the producer.
+
+=cut
+
+sub skip_protected_field ($product_ref, $field, $moderator = 0) {
```

Comment:
It's really cool to have the logic isolated here.

Normally we give protection for only 1 year, couldn't we add the logic here ?

Reply:
It's a bit tricky, if the product has not changed in a year, it does not mean the data is out of date. My preference is to say that we give protection for as long as the information is up to date. If we have evidence the data is not up to date, even after 6 months, then the protection could be removed.

Reply:
Ok, but right now we have no way of removing that protecting isn't it ? (we should at least open an issue…)

Reply:
ok I see you did open the issue.